modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/WFSFeatureSource.java

```diff
@@ -214,9 +234,8 @@ protected GetFeatureRequest createGetFeature(Query query, ResultType resultType)
 
         request.setTypeName(remoteTypeName);
         request.setFullType(remoteSimpleFeatureType);
-
-        invertAxisInFilterIfNeeded(query);
-
+        removeFastBBOXFilterIfNeeded(query);
```

Comment:
A non hacky alternative might be (untested):

```
if (query.getFilter() != null) {
            DuplicatingFilterVisitor duplicator = new DuplicatingFilterVisitor();
            Filter copy = (Filter) query.getFilter().accept(duplicator, null);
            query.setFilter(copy);
        }
```

Tests are still passing using this method instead of the double inversion.

Reply:
Oh, another possible approach would be to implement getSRS in fastBBOX like it's done in BBOX3DImpl: return CRS.toSRS(envelope.getCoordinateReferenceSystem());
Not sure if that will work in all cases, just an idea.